src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ReflectableTypeNode.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+using ILCompiler.DependencyAnalysisFramework;
+
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a type that is visible from reflection.
```

Comment:
Could you please add a comment about the difference between ReflectableTypeNode and TypeMetadataNode?
Naively I would expect that if type is reflectable it needs metadata, but currently it's the other way round - TypeMetadataNode adds ReflectableTypeNode. So that must mean that there's a case where type is reflectable but we don't have metadata for it.